load-test-framework/src/main/java/com/google/pubsub/flic/Driver.java

```diff
@@ -208,29 +239,122 @@ private void run() {
-      log.info("99%: " + LatencyDistribution.getNthPercentile(stats.bucketValues, 99.0));
-      log.info("99.9%: " + LatencyDistribution.getNthPercentile(stats.bucketValues, 99.9));
-      // CPS Publishers report latency per batch message.
-      log.info("Average throughput: " +
-          new DecimalFormat("#.##").format(
-              (double) LongStream.of(
-                  stats.bucketValues).sum() / stats.runningSeconds * messageSize / 1000000.0
-                  * (type.isCpsPublisher() ? cpsPublishBatchSize : 1)) + " MB/s");
+      String[] row = new String[5];
+      row[0] = type.toString();
+      row[1] = LatencyDistribution.getNthPercentile(stats.bucketValues, 50.0);
+      row[2] = LatencyDistribution.getNthPercentile(stats.bucketValues, 99.0);
+      row[3] = LatencyDistribution.getNthPercentile(stats.bucketValues, 99.9);
+      row[4] = new DecimalFormat("#.##").format(
+          (double) LongStream.of(
+              stats.bucketValues).sum() / stats.runningSeconds * messageSize / 1000000.0
+              * (type.isCpsPublisher() ? cpsPublishBatchSize : 1));
     });
+    return out;
+  }
+
+  private void printStats(List<String[]> stats) {
+    for(String[] row : stats) {
+      log.info("Results for " + row[0] + ":");
+      log.info("50%: " + row[1]);
+      log.info("99%: " + row[2]);
+      log.info("99.9%: " + row[3]);
+      // CPS Publishers report latency per batch message.
+      log.info("Average throughput: " + row[4] + " MB/s");
+    }
+  }
+  
+  private Sheets getSheetsService() throws Exception {
+    InputStream in = new FileInputStream(new File(System.getenv("GOOGLE_OATH2_CREDENTIALS")));
+    JsonFactory factory = new JacksonFactory();
+    GoogleClientSecrets clientSecrets = GoogleClientSecrets.load(factory, new InputStreamReader(in));
+    HttpTransport transport = GoogleNetHttpTransport.newTrustedTransport(); 
+    FileDataStoreFactory dataStoreFactory = new FileDataStoreFactory(new File(DATA_STORE_DIR));
+    List<String> scopes = Arrays.asList(SheetsScopes.SPREADSHEETS);
+    GoogleAuthorizationCodeFlow flow =
+        new GoogleAuthorizationCodeFlow.Builder(
+                transport, factory, clientSecrets, scopes)
+        .setAccessType("offline")
+        .setDataStoreFactory(dataStoreFactory)
+        .build();
+    Credential credential = new AuthorizationCodeInstalledApp(
+        flow, new LocalServerReceiver()).authorize("user");
+    return new Sheets.Builder(transport, factory, credential).setApplicationName(APPLICATION_NAME).build();
+  }
+  
+  private void sendToSheets(Sheets sheetService, List<String[]> stats) {
+    List<List<Object>> cpsValues = new ArrayList<List<Object>>();
+    List<List<Object>> kafkaValues = new ArrayList<List<Object>>();
+    for (String[] row : stats) {
+      List<Object> valueRow = new ArrayList<Object>(13);
+      switch (row[0]) {
+        case "cps-gcloud-publisher":
```

Comment:
This shouldn't operate on a List<String[]> this should operate on the entire Map, and you should use enum switching, not String for type safety.